packages/shared/src/types/config/index.ts

```diff
@@ -107,10 +107,15 @@ export type NormalizedConfig = DeepReadonly<{
   provider?: unknown;
   _privateMeta?: RsbuildConfigMeta;
   environments?: {
-    [name: string]: EnvironmentConfig;
+    [name: string]: DeepReadonly<EnvironmentOption>;
   };
 }>;
 
+/** The normalized Rsbuild environment config. */
+export type EnvironmentConfig = Omit<NormalizedConfig, 'environments'> & {
```

Comment:
`EnvironmentConfig` and `EnvironmentOption` is confusing.

I prefer using `EnvironmentConfig` and `NormalizedEnvironmentConfig`.

Reply:
done